Assets/MixedRealityToolkit-SDK/Features/Diagnostics/DiagnosticsHandler.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Core.EventDatum.Boundary;
+using Microsoft.MixedReality.Toolkit.Core.Interfaces.Diagnostics;
+using System;
+using System.Text;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.SDK.DiagnosticsSystem
+{
+    public class DiagnosticsHandler : MonoBehaviour, IMixedRealityDiagnosticsHandler
+    {
+        private bool showCpu;
+        private bool ShowCpu
+        {
+            get { return showCpu; }
+            set
+            {
+                if (showCpu != value)
+                {
+                    showCpu = value;
+                    if (!showCpu)
+                    {
+                        cpuUseTracker.Reset();
+                    }
+                }
+            }
+        }
+
+        private bool ShowFps { get; set; }
+        private bool ShowMemory { get; set; }
+
+        private string displayText;
+        private int numberOfLines = 0;
+        private bool isShowingInformation;
+
+        private CpuUseTracker cpuUseTracker = new CpuUseTracker();
+        private MemoryUseTracker memoryUseTracker = new MemoryUseTracker();
+        private FpsUseTracker fpsUseTracker = new FpsUseTracker();
+
+        public void OnDiagnosticSettingsChanged(DiagnosticsEventData eventData)
+        {
+            this.ShowCpu = eventData.ShowCpu;
+            this.ShowMemory = eventData.ShowMemory;
+            this.ShowFps = eventData.ShowFps;
+            this.enabled = eventData.Visible;
+        }
+
+        void UpdateIsShowingInformation()
+        {
+            isShowingInformation = ShowCpu ||
+                                   ShowFps ||
+                                   ShowMemory;
+        }
+        
+        void Update()
+        {
+            UpdateIsShowingInformation();
+
+            if (!isShowingInformation)
+            {
+                return;
+            }
+
+            StringBuilder sb = new StringBuilder();
```

Comment:
We're making a new string builder on each update?

Reply:
Just not sure we should have any allocations in update loops.

I think keeping the same string builder around and just clearing it in each update loop is a bit better. Idk, I suppose we'd have test the comparison.

Reply:
Ah, I see. I haven't looked at StringBuilder in a while, they added Clear in 4.0 that should do the right thing. Before, setting length would sort of do it... but it still might allocate a new string, so it wasn't guaranteed. I'll Move to a member and call Clear()